modules/server/src/main/scala/io/chrisdavenport/ember/server/EmberServerBuilder.scala

```diff
@@ -0,0 +1,128 @@
+package io.chrisdavenport.ember.server
+
+import cats._
+import cats.implicits._
+import cats.effect._
+import fs2.concurrent._
+import org.http4s._
+import org.http4s.server.Server
+import scala.concurrent.duration._
+import java.net.InetSocketAddress
+import java.util.concurrent.Executors
+import java.nio.channels.AsynchronousChannelGroup
+
+
+final class EmberServerBuilder[F[_]: Concurrent: Timer: ContextShift] private (
+  private val host: String,
+  private val port: Int,
+  private val httpApp: HttpApp[F],
+  private val agR: Resource[F, AsynchronousChannelGroup],
```

Comment:
This is awkward.  I'm not fond of passing `Resource`s or effects as config, but it makes it hard to create a builder without it.  A way around it is to make it a required field in the companion `apply`, but that has kind of crap ergonomics.